server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/ClusterBalanceStatsTests.java

```diff
@@ -300,18 +372,45 @@ private static ClusterState createClusterState(List<DiscoveryNode> nodes, List<T
             .build();
     }
 
-    private static DesiredBalance createDesiredBalance(ClusterState state) {
+    private static DesiredBalance createDesiredBalanceWithEmptyNodeWeights(ClusterState state) {
+        return createDesiredBalance(state, randomDoubleBetween(-1, 1, true), true);
```

Comment:
Nit: seems wasteful to randomize given its unused
```suggestion
        return createDesiredBalance(state, 0, true);
```